### PR TITLE
Completed assessment

### PR DESCRIPTION
**Pull request by jennerdulce**
# Submitting Week 2 Assessment

### ✅ Completed Files
Mark the following as confirmation that the files in this repository are complete.
- [x] Coding Challenges
- [x] Interview Questions
- [x] React Debugging

### 📝 Feedback
If you would like specific feedback on a particular problem please leave details here:
- It was hard to find any resources for 'this'. It seems that all the explanations were the same stating that 'this' refers to an object and thats about it.

### CODE REVIEW

File: interview-questions.md [hunk 1]
```diff
@@ -9,59 +9,82 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is `this` in JavaScript?
 
-  Your answer:
+  Your answer: They keyword `this` pertains to a specific object that is instantiated from a class. Using the keyword `this` refers to the object and the object's own unique properties that may or may not be similar to other objects instantiated from the same class.
 
-  Researched answer:
+  Researched answer: `this` is a JavaScript keyword that references an object. Specifically an object that is derived from a certain class.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:14]: Make sure you are not copying and pasting but you are correct 'this' refers to the object it is inside of. The word that helps this concept along is instantiation. This refers to an object that is an instance of a class. 'This' is dynamic where each object will have some kind of identifier.

Comment (WalkerTexas-coder) [reply interview-questions.md:14]: 'This' operates like a parameter.

File: code-challenges.test.js [hunk 2]
```diff
@@ -15,42 +15,166 @@
...
+    let sequence = []
+    for(let i = 0; i < length; i++){
+        if(i <= 1 ){
+            sequence.push(1)
+        } else {
+            sequence.push(sequence[i - 1] + sequence[i - 2])
+        }
+    }
+    return sequence
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:62]: Excellent work glad we went over this one in class!

File: code-challenges.test.js [hunk 3]
```diff
@@ -15,42 +15,166 @@
...
+        if(index === 0){
+            previousNumber = value
+            return value
+        } else {
+            addedValues = value + previousNumber
+            previousNumber = addedValues
+            return addedValues
+        }
+    })
+}
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:180]: Well done! Highly readable and great logical flow.

File: interview-questions.md [hunk 4]
```diff
@@ -9,59 +9,82 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: yarn is a software that acts as a package manager. The purpose of a package manage is to be able to install packages which contain useful code snippets and JavaScript code that has been made from other developers. Other developers are able to utilize these packages which can greatly expedite development. One of the main use cases I've used yarn for is to create React applications. With just one terminal command, yarn install all of the dependencies needed to create a React application. Running this command creates some files such as: node_modules, package.json, and yarn.lock
 
+  Researched answer: Yarn is a package manager for your code that allows you to use and share JavaScript code with other developers. Yarn offers developers' solutions to different problems, making it easier for users to develop their software.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:51]: Remember these answers need to be in your words. I appreciate that you cite your resources but you won't be doing that in an interview. 

The goal is that you are internalizing what you have researched not showing me your google prowess. Last time I accept something this close to the source material.

File: react-debugging.html [hunk 5]
```diff
@@ -79,9 +80,13 @@ <h1>Draw a Card</h1>
                 )
               })}
             </div>
+            </>
           )
         }
       }
+      // 1. React Fragment
+      // 2. onClick missing this.drawCard
+      // 3. missing this to shuffle deck
```
Comment (WalkerTexas-coder) [inline react-debugging.html:89]: Well done!

**Review by WalkerTexas-coder — APPROVED** [review]
Jenner great work! I need you to check out my feedback below regarding the interview answers. Good work otherwise! Approved!

File: interview-questions.md (hunk 4 above)
Comment (jennerdulce) [reply interview-questions.md:51]: Sorry I guess I misunderstood what the researched answer should entail. I thought I was supposed to have the more concise answer. But I think I remember you saying now that the researched answer should be a revision of 'your answer' portion

**Pull request merged**